clang/tools/scan-build-py/lib/libscanbuild/intercept.py

```diff
@@ -294,9 +294,11 @@ def abspath(cwd, name):
             file_path = abspath(exec_trace['directory'], preprocess_source)
             logging.debug('preprocess_file: %s,', file_path)
             preproess_output_files.add(file_path)
-
         for source in compilation.files:
-            command = [compiler, "-c"] + compilation.flags + [source]
+            command = [compiler]
+            if('-o' not in compilation.flags):
```

Comment:
Option `-o` is not related to option `-c`.
Please ask @tomflinda for more details.

Reply:
@daiyaan-ahmed6  as discussed  with you and @ShengchenJ, this PR is not needed, pls close it.